simpleocr/files.py

```diff
@@ -10,80 +9,165 @@
 GROUND_EXTENSIONS_DEFAULT = GROUND_EXTENSIONS[0]
 
 
-def open_img(path_or_name):
-    """
-    Fuzzy finds a image file given a absolute or relative path, or a name.
-    The name might have no extension, or be in the DATA_DIRECTORY
-    """
-    try_img_ext = functools.partial(try_extensions, IMAGE_EXTENSIONS)
-    data_dir_path = os.path.join(DATA_DIRECTORY, path_or_name)
-    path = path_or_name
-    if not os.path.exists(path):
-        # proceed even when there's no result. ImageFile decides on the exception to raise
-        path = try_img_ext(path_or_name) or try_img_ext(data_dir_path) or path
-    return ImageFile(path)
-
 def try_extensions(extensions, path):
-    """checks if various extensions of a path exist"""
+    """ Checks for various extensions if a path exist if the extension is appended """
     for ext in [""] + extensions:
         if os.path.exists(path + ext):
             return path + ext
     return None
 
 
-class GroundFile(object):
-    """A file with ground truth data about a image (i.e.: characters and their position)"""
-    def __init__(self, path):
+def open_image(path):
+    return ImageFile(get_file_path(path))
+
+
+def get_file_path(path, ground=False):
+    """
+    Get the absolute path for an image or ground file. Valid inputs are:
+    - Relative path with or without extension
+    - File in DATA_DIRECTORY with or without extension
+    - Absolute path with or without extension
+    :param path: image path in str type
+    :param ground: whether the file must be a ground file
+    :return: The absolute path to the file requested
+    """
+    extensions = GROUND_EXTENSIONS if ground else IMAGE_EXTENSIONS
+    # If the path exists, return the path, but make sure it's an absolute path first
+    if os.path.exists(path):
+        return os.path.abspath(path)
+    # Try to find the file with the passed path with the various extensions
+    image_with_extension = try_extensions(extensions, os.path.splitext(path)[0])
+    if image_with_extension:
+        return os.path.abspath(image_with_extension)
+    # The file must be in the data directory if it has not yet been found
+    image_basename = os.path.basename(path)
```

Comment:
It is indeed true that if the file specified is in a subfolder of the data directory, the current function will not support it.